src/lib/agent/Agent.ts

```diff
@@ -193,6 +206,27 @@ export class Agent {
     return this.agentConfig.inboundConnection;
   }
 
+  async returnWhenIsConnected(verkey: Verkey) {
+    const connectionRecord = await poll(
+      () => this.findConnectionByVerkey(verkey),
+      (c: ConnectionRecord) => c.state !== ConnectionState.COMPLETE,
+      100
+    );
+    return connectionRecord;
+  }
```

Comment:
Maybe we could use the event emitter to subscribe for this event? I think that would be a clear approach

Reply:
I started with this approach, but it occurred to be more complicated than I expected.
- We would need to remove a listener afterward, even if it's there for the short life of tests, it could direct someone to wrong solution if we wouldn't do that.
- It mixes async and callbacks together.
- I realized that consuming API via event emitter should be only optional and it's good to have tests demonstrating that. I'm open to adding other tests demonstrating the use of event emitter, but I would also consider showing that in an example instead.